OWML.ModHelper.Logging/ModSocketOutput.cs

```diff
@@ -0,0 +1,55 @@
+﻿using System;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using OWML.Common;
+
+namespace OWML.ModHelper.Logging
+{
+    public class ModSocketOutput : ModConsole
+    {
+        private const string LocalHost = "127.0.0.1";
+
+        private static Socket _socket;
+
+        public ModSocketOutput(IOwmlConfig config, IModLogger logger, IModManifest manifest) : base(config, logger, manifest)
+        {
+            if (_socket != null)
+            {
+                return;
+            }
```

Comment:
Maybe you can do this instead:
`if (_socket == null) 
{
    _socket = CreateSocket();
}`
and put the below code into CreateSocket ?